src/bp-forums/functions.php

```diff
@@ -854,6 +854,28 @@ function bbp_forum_recursive_group_id( $forum_id ) {
 
 	return false;
 }
+
+/**
+ * Check if there's a parent forum that is associated with private group
+ * 
+ *
+ * @param int $forum_id
+ *
+ * @since Buddyboss [BBVERSION]
+ * @return bool
+ */
+function bb_forum_has_parent_with_private_group( $forum_id ) {
+	
+	if ( ! bp_is_active( 'groups' ) ) {
+		return false;
+	}	
+
+	$group_id	= bbp_forum_recursive_group_id( $forum_id );
+	$group		= groups_get_group( $group_id );
+
+	return $group && $group->status === bbp_get_private_status_id();
+}
```

Comment:
```suggestion
/**
 * Check if there's a parent forum that is associated with private group.
 *
 * @since Buddyboss [BBVERSION]
 *
 * @param int $forum_id Forum id.
 *
 * @return bool
 */
function bb_forum_has_parent_with_private_group( $forum_id ) {
	
	if ( ! bp_is_active( 'groups' ) ) {
		return false;
	}

	$group_id = bbp_forum_recursive_group_id( $forum_id );
	if ( empty( $group_id ) ) {
		$group_id = current( bbp_get_forum_group_ids( $forum_id ) );
	}

	$group = groups_get_group( $group_id );

	return $group && $group->status === bbp_get_private_status_id();
}
```

Reply:
@jitendrabanjara1991 as per request changes have been completed. please review it.